pkg/controller/node/node_controller.go

```diff
@@ -531,73 +545,73 @@ func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result,
 	portList, err := listPortsByAttachmentIds(nsxClients, attachment_ids)
 	if err != nil {
 		cachedNodeSet[nodeName] = &statusmanager.NodeStatus{
-			Address: nodeAddress,
+			Addresses: nodeAddresses,
 			Success: false,
 			Reason:  fmt.Sprintf("Error while achieving ports for node %s: %v", nodeName, err),
 		}
 		r.status.SetFromNodes(cachedNodeSet)
 		return reconcile.Result{}, err
 	}
-	lsp, err := filterPortByNodeAddress(nsxClients, portList, nodeAddress)
+	lsps, err := filterPortsByNodeAddresses(nsxClients, portList, nodeAddresses)
 	if err != nil {
 		cachedNodeSet[nodeName] = &statusmanager.NodeStatus{
-			Address: nodeAddress,
+			Addresses: nodeAddresses,
 			Success: false,
 			Reason:  fmt.Sprintf("Error while achieving port with specific address for node %s: %v", nodeName, err),
 		}
 		r.status.SetFromNodes(cachedNodeSet)
 		return reconcile.Result{}, err
 	}
-	reqLogger.Info("Got the port", "port.Id", lsp.Id, "cluster", cluster)
-
-	foundNodeTag := false
-	foundClusterTag := false
 	nodeNameScope := "ncp/node_name"
 	clusterScope := "ncp/cluster"
-	for _, tag := range lsp.Tags {
-		if tag.Scope == nodeNameScope && tag.Tag == request.Name {
-			foundNodeTag = true
-		} else if tag.Scope == clusterScope && tag.Tag == cluster {
-			foundClusterTag = true
+	anyUpdate := false
+	for _, lsp := range(lsps) {
```

Comment:
Does it work for NCP if you tag multiple LSP with the same node name?

Reply:
Synced offline and the new patch set added the check